server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1159,13 +1226,13 @@ protected void sendApplyCommit(DiscoveryNode destination, ApplyCommitRequest app
     }
 
     // TODO: only here temporarily for BWC development, remove once complete
-    public static Settings.Builder addZen1Attribute(Settings.Builder builder) {
-        return builder.put("node.attr.zen1", true);
+    public static Settings.Builder addZen1Attribute(boolean isZen1Node, Settings.Builder builder) {
```

Comment:
what's the reason for changing this method here?

Reply:
The rolling restart requires us to be able to change a node from Zen1 to Zen2 simply by returning a `Settings`, but with the original implementation we had to remove the `node.attr.zen1` setting which isn't possible with this API.

Reply:
ah, I see. Thanks for the explanation.